wp-content/themes/core/components/blocks/buttons/buttons.php

```diff
@@ -1,19 +1,36 @@
 <?php declare(strict_types=1);
 
 use Tribe\Project\Templates\Components\blocks\buttons\Buttons_Block_Controller;
+use Tribe\Project\Templates\Components\link\Link_Controller;
 
 /**
  * @var array $args Arguments passed to the template
  */
 $c = Buttons_Block_Controller::factory( $args );
 
-if ( empty( $c->get_buttons() ) ) {
+if ( empty( $c->get_buttons()->count() ) ) {
 	return;
 }
 ?>
 
 <div <?php echo $c->get_classes(); ?> <?php echo $c->get_attrs(); ?>>
-	<?php foreach ( $c->get_buttons() as $button ) { ?>
-		<?php get_template_part( 'components/link/link', null, $button ); ?>
-	<?php } ?>
+	<?php foreach ( $c->get_buttons() as $button ) {
+		if ( ! $button->cta->link->url ) {
+			continue;
+		}
+
+		$style = $button->button_style ? sanitize_html_class( sprintf( 'b-style__%s', $button->button_style ) ) : '';
+
+		get_template_part( 'components/link/link', '', [
+			Link_Controller::URL            => $button->cta->link->url,
```

Comment:
This is interesting, obviously it would then couple the Field Model implementation to the component and make things less modular. On the other hand, some kind of `Link_Controller::create_from_cta( Cta $cta ): array;` or something like that could be useful, but that the end of the day I believe the CSS classes on all of these are different based on the component they are used in.